components/entitlement/org.wso2.carbon.identity.entitlement/src/test/java/org/wso2/carbon/identity/entitlement/persistence/PAPStatusDataHandlerTest.java

```diff
@@ -0,0 +1,330 @@
+    static final String SUBSCRIBER_ID_DISPLAY_NAME = "Subscriber Id";
+
+    protected static final String ABOUT_POLICY = "POLICY";
+    protected static final String ABOUT_SUBSCRIBER = "SUBSCRIBER";
+    protected static final String POLICY_KEY = "simple_policy";
+    protected static final String SUBSCRIBER_KEY = "PDP_Subscriber_test";
+    protected static final String PAP_POLICY_STORE_TARGET = "PAP POLICY STORE";
+    protected static final String TARGET_ACTION_PERSIST = "PERSIST";
+    protected static final String TARGET_ACTION_REMOVE = "REMOVE";
+    protected static final String TARGET_ACTION_LOAD = "LOAD";
+    protected static final String POLICY_VERSION_1 = "1";
+    protected static final String SAMPLE_USER = "admin";
+
+    StatusHolder statusHolderForAddPolicy;
+    StatusHolder statusHolderForGetPolicy;
+    StatusHolder statusHolderForRemovePolicy;
+
+    StatusHolder statusHolderForAddPolicyInPDP;
+    StatusHolder statusHolderForUpdatePolicyInPDP;
+    StatusHolder statusHolderForRemovePolicyInPDP;
+
+    List<StatusHolder> policyStatusHolderList1;
+    List<StatusHolder> policyStatusHolderList2;
+    List<StatusHolder> policyStatusHolderList3;
+
+    List<StatusHolder> pdpStatusHolderList1;
+    List<StatusHolder> pdpStatusHolderList2;
+    List<StatusHolder> pdpStatusHolderList3;
+
+    PublisherDataHolder publisherDataHolder;
+
+    PAPStatusDataHandler papStatusDataHandler;
+    SubscriberPersistenceManager subscriberPersistenceManager;
+
+    @BeforeMethod
+    public void setUp() throws Exception {
+
+        Properties engineProperties = new Properties();
+        engineProperties.put(PDPConstants.MAX_NO_OF_STATUS_RECORDS, "5");
+        EntitlementConfigHolder.getInstance().setEngineProperties(engineProperties);
+
+        Properties storeProps = new Properties();
+        storeProps.put(PDPConstants.MAX_NO_OF_STATUS_RECORDS, "5");
+        papStatusDataHandler = createPAPStatusDataHandler();
+        papStatusDataHandler.init(storeProps);
+
+        subscriberPersistenceManager = createSubscriberPersistenceManager();
+        setupPolicyStatusData();
+        setupSubscriberStatusData();
+    }
+
+    @AfterMethod
+    public void tearDown() throws Exception {
+
+        papStatusDataHandler.handle(ABOUT_POLICY, POLICY_KEY, policyStatusHolderList3);
+        papStatusDataHandler.handle(ABOUT_SUBSCRIBER, SUBSCRIBER_KEY, pdpStatusHolderList3);
+        subscriberPersistenceManager.removeSubscriber(SUBSCRIBER_KEY);
+    }
+
+    @Test(priority = 1)
+    public void testHandlePolicyStatus() throws Exception {
+
+        papStatusDataHandler.handle(ABOUT_POLICY, POLICY_KEY, policyStatusHolderList1);
+
+        StatusHolder[] statusDataFromStorage =
+                papStatusDataHandler.getStatusData(ABOUT_POLICY, POLICY_KEY, ADD_POLICY, "*");
+        assertEquals(statusDataFromStorage.length, 1);
+        assertEquals(statusDataFromStorage[0].getType(), statusHolderForAddPolicy.getType());
```

Comment:
here we add `policyStatusHolderList1` but validate the value against a different object `statusHolderForAddPolicy`.. one should have to have the understanding of the whole test class to correlate whats the relation between `policyStatusHolderList1` & `statusHolderForAddPolicy`

Reply:
Addressed with https://github.com/wso2/carbon-identity-framework/compare/e393b967246d88953a960b77a0c4a7c355fd9dfd..f26be27fb80abcc3adb9aace8db0909c0eb3c17d